weave/index/index_sql.py

```diff
@@ -0,0 +1,886 @@
+                              "'sqlalchemy' are required to use this class")
+
+        # Check that the required environment variables are set.
+        try:
+            # We want to fail early if these are not set, so we check them here
+            # even though they are not used until we connect.
+            # Pylint thinks this is pointless, so pylint is getting ignored.
+            # pylint: disable=pointless-statement
+            os.environ["MSSQL_HOST"]
+            os.environ["MSSQL_USERNAME"]
+            os.environ["MSSQL_PASSWORD"]
+            # pylint: enable=pointless-statement
+        except KeyError as key_error:
+            raise KeyError("The following environment variables must be set "
+                           "to use this class: MSSQL_HOST, MSSQL_USERNAME, "
+                           "MSSQL_PASSWORD.") from key_error
+
+        self._file_system = file_system
+        self._pantry_path = pantry_path
+
+        # Set the database name (defaults to weave_db). DATABASE MUST ALREADY
+        # EXIST. If it does not, the user must create it manually.
+        self._database_name = kwargs.get("database_name", "weave_db")\
+            .replace("-", "_")
+
+        # Set the schema name (defaults to pantry_path). If the schema does not
+        # exist, it will be created.
+        d_schema_name = self._pantry_path.replace(os.sep, "_")\
+            .replace("-", "_")
+        self._pantry_schema = kwargs.get("pantry_schema", d_schema_name)
+
+        self._engine = sqla.create_engine(sqla.engine.url.URL(
+            drivername="mssql+pyodbc",
+            username=os.environ["MSSQL_USERNAME"],
+            password=os.environ["MSSQL_PASSWORD"],
+            host=os.environ["MSSQL_HOST"],
+            database=self.database_name,
+            query={
+                "driver": kwargs.get("odbc_driver",
+                                     "ODBC Driver 18 for SQL Server"),
+                "Encrypt": "yes" if kwargs.get("encrypt", False) else "no",
+            },
+            port=os.environ.get("MSSQL_PORT", "1433"),
+        ))
+
+        self._create_schema()
+        self._create_tables()
+
+    # Create read-only properties for the database and schema names.
+    @property
+    def database_name(self):
+        """The database name of the index."""
+        return self._database_name
+
+    @property
+    def pantry_schema(self):
+        """The schema name of the index."""
+        return self._pantry_schema
+
+    def execute_sql(self, sql_query, params=None, commit=False):
+        """Executes the given SQL query. Returns the results.
+
+        Parameters
+        ----------
+        sql_query: str or sqlalchemy.sql.text
+            The SQL query to be executed.
+        params: dict (optional)
+            The parameters to be used in the query.
+        commit: bool (default=False)
+            Whether or not to commit the query.
```

Comment:
I don't understand why the commit flag is here? If we are calling "execute_sql" wouldn't we always want to commit? What scenarios would we want to run the query and not commit changes?